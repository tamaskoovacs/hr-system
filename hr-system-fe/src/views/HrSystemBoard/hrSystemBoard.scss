.top-section{
    text-align: center;
    margin-bottom: 20px;
}