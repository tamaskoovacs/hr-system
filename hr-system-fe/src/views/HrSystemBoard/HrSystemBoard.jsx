import Table from "components/Table/Table"
import tableData from 'static/tableData.json'
import { useNavigate,  } from 'react-router-dom'
import './hrSystemBoard.scss'

const HrSystemBoard = () => {
    let navigate = useNavigate();
    const handleChartsClick = () => {
        navigate('/charts')
    }
    return (
        <>
            <div className="top-section">
                <h1>Hr System Board</h1>
                <button onClick={handleChartsClick}>Charts</button>
            </div>
            <Table data={tableData} headers={tableHeaders} pageSize={20}/>
        </>
    )
}

const tableHeaders = ['Id', 'First Name', 'Last Name', 'Email', 'Date of birth', 'Industry', 'Salary', 'Years of experience', 'Actions']

export default HrSystemBoard;