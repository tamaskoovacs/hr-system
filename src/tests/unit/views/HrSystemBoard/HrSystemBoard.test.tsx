import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import HrSystemBoard from 'views/HrSystemBoard/HrSystemBoard';
import { BrowserRouter } from 'react-router-dom';

describe('HrSystemBoard page', () => {
  test('It should matches snapshot', () => {
    const view = render(
      <BrowserRouter>
        <HrSystemBoard />
      </BrowserRouter>
    );
    expect(view.container).toMatchSnapshot();
  });

  test('It should navigate to the Charts page after button click', () => {
    render(
      <BrowserRouter>
        <HrSystemBoard />
      </BrowserRouter>
    );
    fireEvent.click(screen.getByText('Charts'));
    expect(document.location.pathname).toBe('/charts');
  });
});
