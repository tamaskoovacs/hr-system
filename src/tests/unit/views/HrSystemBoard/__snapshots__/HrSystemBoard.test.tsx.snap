// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HrSystemBoard page It should matches snapshot 1`] = `
<div>
  <div
    class="top-section"
  >
    <h1>
      Hr System Board
    </h1>
    <button>
      Charts
    </button>
  </div>
  <form>
    <table>
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            First Name
          </th>
          <th>
            Last Name
          </th>
          <th>
            Email
          </th>
          <th>
            Date of birth
          </th>
          <th>
            Industry
          </th>
          <th>
            Salary
          </th>
          <th>
            Years of experience
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            data-label="Id"
          >
            1
          </td>
          <td
            data-label="First Name"
          >
            Loutitia
          </td>
          <td
            data-label="Last Name"
          >
            Steaning
          </td>
          <td
            data-label="Email"
          >
            lsteaning0@usnews.com
          </td>
          <td
            data-label="Date of birth"
          >
            13/05/1978
          </td>
          <td
            data-label="Industry"
          >
            n/a
          </td>
          <td
            data-label="Salary"
          >
            98803.83
          </td>
          <td
            data-label="Years of experience"
          >
            6.6
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-1"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-1"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            2
          </td>
          <td
            data-label="First Name"
          >
            Ewen
          </td>
          <td
            data-label="Last Name"
          >
            McLewd
          </td>
          <td
            data-label="Email"
          >
            emclewd1@bbb.org
          </td>
          <td
            data-label="Date of birth"
          >
            15/12/1991
          </td>
          <td
            data-label="Industry"
          >
            Telecommunications Equipment
          </td>
          <td
            data-label="Salary"
          >
            144392.9
          </td>
          <td
            data-label="Years of experience"
          >
            2.8
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-2"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-2"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            3
          </td>
          <td
            data-label="First Name"
          >
            Park
          </td>
          <td
            data-label="Last Name"
          />
          <td
            data-label="Email"
          />
          <td
            data-label="Date of birth"
          >
            11/06/1993
          </td>
          <td
            data-label="Industry"
          >
            n/a
          </td>
          <td
            data-label="Salary"
          >
            101773.01
          </td>
          <td
            data-label="Years of experience"
          >
            1.8
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-3"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-3"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            4
          </td>
          <td
            data-label="First Name"
          >
            Benson
          </td>
          <td
            data-label="Last Name"
          >
            Possek
          </td>
          <td
            data-label="Email"
          >
            bpossek3@devhub.com
          </td>
          <td
            data-label="Date of birth"
          >
            01/03/1993
          </td>
          <td
            data-label="Industry"
          >
            n/a
          </td>
          <td
            data-label="Salary"
          >
            141822.61
          </td>
          <td
            data-label="Years of experience"
          >
            3.7
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-4"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-4"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            5
          </td>
          <td
            data-label="First Name"
          >
            Derril
          </td>
          <td
            data-label="Last Name"
          >
            Santorini
          </td>
          <td
            data-label="Email"
          >
            dsantorini4@wp.com
          </td>
          <td
            data-label="Date of birth"
          >
            10/10/1991
          </td>
          <td
            data-label="Industry"
          >
            Electrical Products
          </td>
          <td
            data-label="Salary"
          >
            144774.96
          </td>
          <td
            data-label="Years of experience"
          >
            4.1
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-5"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-5"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            6
          </td>
          <td
            data-label="First Name"
          >
            Lisa
          </td>
          <td
            data-label="Last Name"
          >
            Cobb
          </td>
          <td
            data-label="Email"
          >
            lcobb5@trellian.com
          </td>
          <td
            data-label="Date of birth"
          >
            22/02/1999
          </td>
          <td
            data-label="Industry"
          >
            Major Pharmaceuticals
          </td>
          <td
            data-label="Salary"
          >
            116433.83
          </td>
          <td
            data-label="Years of experience"
          >
            4.3
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-6"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-6"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            7
          </td>
          <td
            data-label="First Name"
          >
            Erek
          </td>
          <td
            data-label="Last Name"
          >
            Corbould
          </td>
          <td
            data-label="Email"
          >
            ecorbould6@blog.com
          </td>
          <td
            data-label="Date of birth"
          >
            10/07/1961
          </td>
          <td
            data-label="Industry"
          >
            Steel/Iron Ore
          </td>
          <td
            data-label="Salary"
          >
            72515.73
          </td>
          <td
            data-label="Years of experience"
          >
            3
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-7"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-7"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            8
          </td>
          <td
            data-label="First Name"
          >
            Stacee
          </td>
          <td
            data-label="Last Name"
          >
            McAdam
          </td>
          <td
            data-label="Email"
          >
            smcadam7@pen.io
          </td>
          <td
            data-label="Date of birth"
          >
            26/04/1995
          </td>
          <td
            data-label="Industry"
          >
            Computer Software: Prepackaged Software
          </td>
          <td
            data-label="Salary"
          >
            79521.66
          </td>
          <td
            data-label="Years of experience"
          >
            2.4
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-8"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-8"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            9
          </td>
          <td
            data-label="First Name"
          >
            Anna-diane
          </td>
          <td
            data-label="Last Name"
          >
            Urquhart
          </td>
          <td
            data-label="Email"
          >
            aurquhart8@reuters.com
          </td>
          <td
            data-label="Date of birth"
          >
            09/10/1968
          </td>
          <td
            data-label="Industry"
          >
            Precious Metals
          </td>
          <td
            data-label="Salary"
          >
            55216.7
          </td>
          <td
            data-label="Years of experience"
          >
            4
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-9"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-9"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            10
          </td>
          <td
            data-label="First Name"
          >
            Goldia
          </td>
          <td
            data-label="Last Name"
          >
            Axelby
          </td>
          <td
            data-label="Email"
          >
            gaxelby9@bizjournals.com
          </td>
          <td
            data-label="Date of birth"
          >
            06/05/1986
          </td>
          <td
            data-label="Industry"
          >
            Business Services
          </td>
          <td
            data-label="Salary"
          >
            58835.13
          </td>
          <td
            data-label="Years of experience"
          >
            6.9
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-10"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-10"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            11
          </td>
          <td
            data-label="First Name"
          >
            Natty
          </td>
          <td
            data-label="Last Name"
          >
            Boschmann
          </td>
          <td
            data-label="Email"
          >
            nboschmanna@geocities.jp
          </td>
          <td
            data-label="Date of birth"
          >
            18/09/1989
          </td>
          <td
            data-label="Industry"
          >
            Military/Government/Technical
          </td>
          <td
            data-label="Salary"
          >
            47565.76
          </td>
          <td
            data-label="Years of experience"
          >
            3.2
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-11"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-11"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            12
          </td>
          <td
            data-label="First Name"
          >
            Gina
          </td>
          <td
            data-label="Last Name"
          >
            Bernollet
          </td>
          <td
            data-label="Email"
          />
          <td
            data-label="Date of birth"
          >
            22/10/1970
          </td>
          <td
            data-label="Industry"
          >
            n/a
          </td>
          <td
            data-label="Salary"
          >
            67862.02
          </td>
          <td
            data-label="Years of experience"
          >
            4.4
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-12"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-12"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            13
          </td>
          <td
            data-label="First Name"
          >
            Carlie
          </td>
          <td
            data-label="Last Name"
          >
            Twiddy
          </td>
          <td
            data-label="Email"
          />
          <td
            data-label="Date of birth"
          >
            13/10/1974
          </td>
          <td
            data-label="Industry"
          >
            n/a
          </td>
          <td
            data-label="Salary"
          >
            77239.33
          </td>
          <td
            data-label="Years of experience"
          >
            2.7
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-13"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-13"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            14
          </td>
          <td
            data-label="First Name"
          >
            Daniel
          </td>
          <td
            data-label="Last Name"
          >
            Commin
          </td>
          <td
            data-label="Email"
          />
          <td
            data-label="Date of birth"
          >
            19/12/1968
          </td>
          <td
            data-label="Industry"
          >
            Investment Bankers/Brokers/Service
          </td>
          <td
            data-label="Salary"
          >
            139549.35
          </td>
          <td
            data-label="Years of experience"
          >
            1.2
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-14"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-14"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            15
          </td>
          <td
            data-label="First Name"
          >
            Robyn
          </td>
          <td
            data-label="Last Name"
          >
            Siaskowski
          </td>
          <td
            data-label="Email"
          >
            rsiaskowskie@admin.ch
          </td>
          <td
            data-label="Date of birth"
          >
            17/06/1966
          </td>
          <td
            data-label="Industry"
          >
            Oil & Gas Production
          </td>
          <td
            data-label="Salary"
          >
            129885.03
          </td>
          <td
            data-label="Years of experience"
          >
            6.7
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-15"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-15"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            16
          </td>
          <td
            data-label="First Name"
          >
            Creight
          </td>
          <td
            data-label="Last Name"
          >
            De Domenicis
          </td>
          <td
            data-label="Email"
          >
            cdedomenicisf@cam.ac.uk
          </td>
          <td
            data-label="Date of birth"
          >
            16/07/1979
          </td>
          <td
            data-label="Industry"
          >
            Electric Utilities: Central
          </td>
          <td
            data-label="Salary"
          >
            29051.5
          </td>
          <td
            data-label="Years of experience"
          >
            2.2
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-16"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-16"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            17
          </td>
          <td
            data-label="First Name"
          >
            Berti
          </td>
          <td
            data-label="Last Name"
          >
            Pietruszka
          </td>
          <td
            data-label="Email"
          />
          <td
            data-label="Date of birth"
          >
            22/07/1970
          </td>
          <td
            data-label="Industry"
          >
            Hospital/Nursing Management
          </td>
          <td
            data-label="Salary"
          >
            37773.8
          </td>
          <td
            data-label="Years of experience"
          >
            6.4
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-17"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-17"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            18
          </td>
          <td
            data-label="First Name"
          >
            Eveline
          </td>
          <td
            data-label="Last Name"
          >
            Girardez
          </td>
          <td
            data-label="Email"
          >
            egirardezh@behance.net
          </td>
          <td
            data-label="Date of birth"
          >
            21/05/1986
          </td>
          <td
            data-label="Industry"
          >
            Computer Software: Prepackaged Software
          </td>
          <td
            data-label="Salary"
          >
            56974.77
          </td>
          <td
            data-label="Years of experience"
          >
            0.1
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-18"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-18"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            19
          </td>
          <td
            data-label="First Name"
          >
            Robinet
          </td>
          <td
            data-label="Last Name"
          >
            O'Bradane
          </td>
          <td
            data-label="Email"
          >
            robradanei@cnn.com
          </td>
          <td
            data-label="Date of birth"
          >
            23/12/1998
          </td>
          <td
            data-label="Industry"
          >
            Real Estate Investment Trusts
          </td>
          <td
            data-label="Salary"
          >
            139551.49
          </td>
          <td
            data-label="Years of experience"
          >
            5.9
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-19"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-19"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td
            data-label="Id"
          >
            20
          </td>
          <td
            data-label="First Name"
          >
            Lari
          </td>
          <td
            data-label="Last Name"
          >
            Mateo
          </td>
          <td
            data-label="Email"
          >
            lmateoj@bravesites.com
          </td>
          <td
            data-label="Date of birth"
          >
            09/12/1974
          </td>
          <td
            data-label="Industry"
          >
            Computer Software: Programming, Data Processing
          </td>
          <td
            data-label="Salary"
          >
            44826.25
          </td>
          <td
            data-label="Years of experience"
          >
            2.9
          </td>
          <td>
            <button
              data-testid="read-only-row-edit-bt-20"
              type="button"
            >
              Edit
            </button>
            <button
              data-testid="read-only-row-delete-bt-20"
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <ul
      class="pagination-container pagination-bar"
    >
      <li
        class="pagination-item disabled"
        data-testid="left-arrow-li"
      >
        <div
          class="arrow left"
        />
      </li>
      <li
        class="pagination-item selected"
        data-testid="pagination-li-1"
      >
        1
      </li>
      <li
        class="pagination-item"
        data-testid="pagination-li-2"
      >
        2
      </li>
      <li
        class="pagination-item"
        data-testid="pagination-li-3"
      >
        3
      </li>
      <li
        class="pagination-item"
        data-testid="pagination-li-4"
      >
        4
      </li>
      <li
        class="pagination-item"
        data-testid="pagination-li-5"
      >
        5
      </li>
      <li
        class="pagination-item dots"
      >
        …
      </li>
      <li
        class="pagination-item"
        data-testid="pagination-li-7"
      >
        25
      </li>
      <li
        class="pagination-item"
        data-testid="right-arrow-li"
      >
        <div
          class="arrow right"
        />
      </li>
    </ul>
  </form>
</div>
`;
